server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1192,50 +1192,45 @@ private Engine.IndexResult index(Engine engine, Engine.Index index) throws IOExc
         active.set(true);
         final Engine.IndexResult result;
         index = indexingOperationListeners.preIndex(shardId, index);
+        final Engine.Index finalIndex = index;
```

Comment:
How about:

`final Engine.Index finalIndex = indexingOperationListeners.preIndex(shardId, index);`

and then use `finalIndex` throughout.

also, do take into account that we may be breaking some things here because in L1234:
` indexingOperationListeners.postIndex(shardId, index, e);` because this operation will need to be operated on finalIndex then.

Reply:
Fixed it